x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/TransformConfigVersion.java

```diff
@@ -0,0 +1,395 @@
+ * A new Transform config version should be added <em>every time</em> a change is made to the serialization format of one or more Transform
+ * config or state classes.
+ * Each Transform config version should only be used in a single merged commit (apart from BwC versions copied from {@link Version}).
+ * <p>
+ * To add a new Transform config version, add a new constant at the bottom of the list that is one million greater than the current highest
+ * version, ensure it has a unique id, and update the {@link #CURRENT} constant to point to the new version.
+ * <h2>Reverting a Transform config version</h2>
+ * If you revert a commit with a Transform config version change, you <em>must</em> ensure there is a <em>new</em> Transform config version
+ * representing the reverted change. <em>Do not</em> let the Transform config version go backwards, it must <em>always</em> be incremented.
+ */
+public record TransformConfigVersion(int id) implements VersionId<TransformConfigVersion>, ToXContentFragment {
+
+    /*
+     * NOTE: IntelliJ lies!
+     * This map is used during class construction, referenced by the registerTransformConfigVersion method.
+     * When all the Transform config version constants have been registered, the map is cleared & never touched again.
+     */
+    private static Map<String, Integer> IDS = new HashMap<>();
+
+    private static TransformConfigVersion registerTransformConfigVersion(int id, String uniqueId) {
+        checkUniqueness(id, uniqueId);
+        return new TransformConfigVersion(id);
+    }
+
+    private static void checkUniqueness(int id, String uniqueId) {
+        if (IDS == null) throw new IllegalStateException("The IDS map needs to be present to call this method");
+
+        Strings.requireNonEmpty(uniqueId, "Each TransformConfigVersion needs a unique string id");
+        Integer existing = IDS.put(uniqueId, id);
+        if (existing != null) {
+            throw new IllegalArgumentException("Versions " + id + " and " + existing + " have the same unique id");
+        }
+    }
+
+    public static final String TRANSFORM_CONFIG_VERSION_NODE_ATTR = "transform.config_version";
+
+    public static final TransformConfigVersion ZERO = registerTransformConfigVersion(0, "00000000-0000-0000-0000-000000000000");
+    public static final TransformConfigVersion V_5_4_0 = registerTransformConfigVersion(5_04_00_99, "B917B447-2CC6-4BD8-BEB2-733D157B7B37");
+    public static final TransformConfigVersion V_6_0_0 = registerTransformConfigVersion(6_00_00_99, "E3F1CC55-A992-4CB3-BCE6-EC310D156793");
+    public static final TransformConfigVersion V_7_0_0 = registerTransformConfigVersion(7_00_00_99, "21906C88-0955-4BE3-B4B3-BFB6B9606112");
+    public static final TransformConfigVersion V_7_0_1 = registerTransformConfigVersion(7_00_01_99, "7FF09B06-20B3-40FF-A6E0-EC94F55B8B56");
+    public static final TransformConfigVersion V_7_1_0 = registerTransformConfigVersion(7_01_00_99, "BD49F37F-1E50-4BC1-B227-EE0AC88B677C");
```

Comment:
If I remember correctly, transforms was first released in 7.2, so these earlier constants can be deleted.